README.md

```diff
@@ -2,11 +2,45 @@
 
 The `firebase-functions-test` is unit testing library for Cloud Functions for Firebase. It is a companion to [firebase-functions](https://github.com/Firebase/firebase-functions).
 
-_NOTE: This library can only be used with `firebase-functions` v2.0.0 or above._
+_NOTE: This library can only be used with `firebase-functions` v3.20.1 or above._
+
+## Usage
+
+1. Write some Firebase Functions
+2. With your testing framework of choice, write a unit-test that imports your Firebase Functions.
+3. `wrap` your Firebase Functions. You can invoke the Firebase Function's handler by invoking the `wrap` call.
+
+Eg.
+
+```typescript
+import {myFirebaseFunction} from "../index"; // Your Firebase Functions
+import firebaseFunctionsTest from "firebase-functions-test";
+
+// Extracting `wrap` out of the lazy-loaded features
+const {wrap} = firebaseFunctionsTest();
+
+// `jest-ts` example
+test('my unit test', () => {
+  const wrappedFirebaseFunction = wrap(myFirebaseFunction);
+
+  // Invoke the firebase function
+  wrappedFirebaseFunction();
+
+  // Invoke the firebase function with CloudEvent overrides
+  wrappedFirebaseFunction({data: {arbitrary: 'values'}});
+});
+```
+
+## Examples
+
+* [Unit Testing Gen-1 Cloud Functions using Mocha](https://github.com/firebase/functions-samples/tree/main/quickstarts/uppercase/functions)
+* [Unit Testing Gen-2 Cloud Functions using Mocha](https://github.com/firebase/functions-samples/tree/main/2nd-gen/test-functions-mocha/functions)
+* [Unit Testing Gen-2 Cloud Functions using Jest](https://github.com/firebase/functions-samples/tree/main/2nd-gen/test-functions-jest/functions)
+* [Unit Testing Gen-2 Cloud Functions using Jest-Ts](https://github.com/firebase/functions-samples/tree/main/2nd-gen/test-functions-jest-ts/functions)
 
 ## Learn more
 
-Learn more about the SDK [Firebase documentation](https://firebase.google.com/docs/functions/unit-testing) or [check out sample unit tests](https://github.com/firebase/functions-samples/blob/master/quickstarts/uppercase/functions).
+Learn more about the SDK [Firebase documentation](https://firebase.google.com/docs/functions/unit-testing).
```

Comment:
Nit: "the SDK Firebase documentation" sounds wonky.

Reply:
Agreed, updated!